cmake/custom/catch.cmake

```diff
@@ -1 +1 @@
-include_directories(SYSTEM ${PROJECT_SOURCE_DIR}/external/Catch)
+include_directories(SYSTEM ${PROJECT_SOURCE_DIR}/external/catch)
```

Comment:
If you are using Catch2, you can actually remove this file and to what is suggested on their docs. Put this in the `CMakeLists.txt` under the `tests` directory:
```
# Prepare "Catch" library for other executables
set(CATCH_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/catch)
add_library(Catch INTERFACE)
target_include_directories(Catch INTERFACE ${CATCH_INCLUDE_DIR})
```
and then link the unit tests executable as:
```
target_link_libraries(tests Catch)
```
The benefit is that everything test-related is localized to where the tests actually are and the rest of the project doesn't have the spurious include in their compiler flags.

Reply:
I will upgrade Catch and do these changes in a separate PR